pkg/services/guardian/accesscontrol_guardian.go

```diff
@@ -225,44 +333,75 @@ func (a *AccessControlDashboardGuardian) CanCreate(folderID int64, isFolder bool
 	return a.evaluate(accesscontrol.EvalPermission(dashboards.ActionDashboardsCreate, dashboards.ScopeFoldersProvider.GetResourceScopeUID(folder.UID)))
 }
 
-func (a *AccessControlDashboardGuardian) evaluate(evaluator accesscontrol.Evaluator) (bool, error) {
+func (a *accessControlFolderGuardian) CanCreate(folderID int64, isFolder bool) (bool, error) {
```

Comment:
Nit: should this guardian deal with the nested folder scenario of checking the permissions to create a folder in a folder? I think we currently do it somewhere in the handler code, but we could move it over to here. I also think that we can remove the `isFolder` flag now that we have both dashboard and folder guardians. But I'm also ok to leave this as it is and clean it up later.

Reply:
> Nit: should this guardian deal with the nested folder scenario of checking the permissions to create a folder in a folder? I think we currently do it somewhere in the handler code, but we could move it over to here. 

I like the suggestion: currently we do it in [the folder service implementation](https://github.com/grafana/grafana/blob/8c8b3ecb5babac17b708ab3268c61556bfb83f1a/pkg/services/folder/folderimpl/folder.go#L255) but I think it's better to move it in the guardian: if you don't mind I will do it a follow up!

> I also think that we can remove the isFolder flag now that we have both dashboard and folder guardians. But I'm also ok to leave this as it is and clean it up later.
I think this is related to @zserge's suggestion: I will check the callers and try to remove it in a follow up

Reply:
Yup, let's do it in a follow up 👍